onecodex/api.py

```diff
@@ -1,31 +1,68 @@
 from __future__ import print_function
-from datetime import datetime
-import errno
 import filelock
 import json
 import logging
 import os
+import requests
 from requests.adapters import HTTPAdapter
 from requests.auth import HTTPBasicAuth
 from requests.packages.urllib3.util.retry import Retry
 import warnings
 
-from onecodex.exceptions import OneCodexException
 from onecodex.lib.auth import BearerTokenAuth
 from onecodex.utils import collapse_user, init_sentry
-from onecodex.vendored.potion_client import Client as PotionClient
-from onecodex.vendored.potion_client.converter import (
-    PotionJSONSchemaDecoder,
-    PotionJSONDecoder,
-    PotionJSONEncoder,
-)
-from onecodex.vendored.potion_client.utils import upper_camel_case
 from onecodex.version import __version__
 
 
 log = logging.getLogger("onecodex")
 
 
+class HTTPClient:
+    """Simple requests-based HTTP client for API requests."""
+
+    def __init__(self, auth=None, headers=None):
+        self.session = requests.Session()
+        if auth:
+            self.session.auth = auth
+        if headers:
+            self.session.headers.update(headers)
+
+        # Set backoff / retry strategy for 429s
+        retry_strategy = Retry(
+            total=3, backoff_factor=4, allowed_methods=None, status_forcelist=[429, 502, 503]
+        )
+        adapter = HTTPAdapter(max_retries=retry_strategy)
+        self.session.mount("http://", adapter)
+        self.session.mount("https://", adapter)
+
+    def get(self, url, **kwargs):
+        """Make GET request."""
+        return self.session.request(url, "GET", **kwargs)
+
+    def post(self, url, **kwargs):
+        """Make POST request."""
+        return self.session.request(url, "POST", **kwargs)
+
+    def patch(self, url, **kwargs):
+        """Make PATCH request."""
+        return self.session.request(url, "PATCH", **kwargs)
+
+    def put(self, url, **kwargs):
+        """Make PUT request."""
+        return self.session.request(url, "PUT", **kwargs)
+
+    def delete(self, url, **kwargs):
+        """Make DELETE request."""
+        return self.session.request(url, "DELETE", **kwargs)
```

Comment:
Should all of these methods be calling `self.request()` instead of `self.session.request()`?

Reply:
Yes